### PR TITLE
feat: organization tags UI (GSoC)

### LINKED ISSUE #2037: ADMIN: Implement Tags UI (GSoC)
### Describe the feature.

Adding support for `Tags` in the admin.

### Describe the solution you'd like

Making the `Tags` UI, adding a `Tags` option in the sidebar that would show us organization tags, as shown in the image below:

Organization_ Tags

### Additional context

This issue is part of the GSoC project `Tag management system`.

### PR DESCRIPTION
**Pull request by meetulr**
### What kind of change does this PR introduce?

Implements Organization Tags UI.

### Issue Number:

Fixes #2037 

### Did you add tests for your changes?

Yes

### Snapshots/Videos

https://github.com/user-attachments/assets/214c4845-e1e4-41ca-b74c-cea8ae2c44f2

### Additional Information

This is the PR for organization tag screen, following this will be screens for managing tags and sub tags.

## Summary by CodeRabbit

- **New Features**
	- Added a dedicated section for managing organization tags, enhancing user interface capabilities.
	- Introduced support for tag management across multiple languages, including English, French, Hindi, Spanish, and Chinese.
	- Enhanced navigation by adding a new route for organization tags.

- **Bug Fixes**
	- Improved error handling for tag-related actions, ensuring better user feedback.

- **Documentation**
	- Updated the CONTRIBUTING.md file with a new section for video resources to guide contributors.

- **Style**
	- Introduced new CSS styles for the organization tags interface, improving the overall user experience.

- **Tests**
	- Added comprehensive unit tests for the OrganizationTags component to ensure reliability and functionality.

### CODE REVIEW

File: src/GraphQl/Mutations/TagMutations.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+//  */
+
+// export const UPDATE_USER_TAG = gql`
+//   mutation UpdateUserTag($tagId: ID!, $name: String!) {
+//     updateUserTag(input: { tagId: $tagId, name: $name }) {
+//       _id
+//     }
+//   }
+// `;
+
```
Comment (beingnoble03) [inline src/GraphQl/Mutations/TagMutations.ts:61]: why is this commented? is it not ready yet or is it no longer required?

File: src/screens/OrganizationTags/OrganizationTags.tsx [hunk 2]
```diff
@@ -0,0 +1,511 @@
...
+      }
+    }
+  };
+
+  if (createUserTagLoading || orgUserTagsLoading) {
+    return <Loader />;
+  }
+
+  if (orgUserTagsError) {
+    console.log(orgUserTagsError.message);
```
Comment (beingnoble03) [inline src/screens/OrganizationTags/OrganizationTags.tsx:154]: i think this `console.log` can be removed?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -9,6 +9,7 @@ If you are new to contributing to open source, please read the Open Source Guide
 <!-- toc -->
 
 - [Code of Conduct](#code-of-conduct)
+- [Videos](#videos)
```
Comment (beingnoble03) [inline CONTRIBUTING.md:12]: Is this change required? Please give some context.

File: src/App.tsx [hunk 4]
```diff
@@ -145,6 +148,15 @@ function app(): JSX.Element {
...
             <Route path="/orgpeople/:orgId" element={<OrganizationPeople />} />
+            <Route path="/orgtags/:orgId" element={<OrganizationTags />} />
+            {/* <Route
+              path="orgtags/:orgId/orgtagdetails/:tagId"
+              element={<OrganizationTagDetails />}
+            />
+            <Route
+              path="/orgtags/:orgId/orgtagChildTags/:tagId"
+              element={<OrganizationTagChildTags />}
+            /> */}
```
Comment (beingnoble03) [inline src/App.tsx:159]: Why is this commented out? Is this something that's planned for future?

File: src/App.tsx [hunk 5]
```diff
@@ -20,6 +20,9 @@ import OrganizationEvents from 'screens/OrganizationEvents/OrganizationEvents';
 import OrganizaitionFundCampiagn from 'screens/OrganizationFundCampaign/OrganizationFundCampagins';
 import OrganizationFunds from 'screens/OrganizationFunds/OrganizationFunds';
 import OrganizationPeople from 'screens/OrganizationPeople/OrganizationPeople';
+import OrganizationTags from 'screens/OrganizationTags/OrganizationTags';
+// import OrganizationTagDetails from 'screens/OrgnanizationTagDetails/OrganizationTagDetails';
+// import OrganizationTagChildTags from 'screens/OrganizationTagChildTags/OrganizationTagChildTags';
```
Comment (beingnoble03) [inline src/App.tsx:25]: Is this part of a future plan?

**Review by beingnoble03 — COMMENTED** [review]
A bit of a nit-pick. Overall, the changes look good.

**Review by beingnoble03 — DISMISSED** [review]

File: src/GraphQl/Mutations/TagMutations.ts (hunk 1 above)
Comment (meetulr) [reply src/GraphQl/Mutations/TagMutations.ts:61]: the commented out code will be used in the upcoming PRs.

File: src/App.tsx (hunk 4 above)
Comment (meetulr) [reply src/App.tsx:159]: the commented out code will be used in the upcoming PRs.

File: src/App.tsx (hunk 5 above)
Comment (meetulr) [reply src/App.tsx:25]: yes

File: src/screens/OrganizationTags/OrganizationTags.tsx (hunk 2 above)
Comment (meetulr) [reply src/screens/OrganizationTags/OrganizationTags.tsx:154]: yes, thank you

File: CONTRIBUTING.md (hunk 3 above)
Comment (meetulr) [reply CONTRIBUTING.md]: was added automatically, I have removed it.

**Comment by beingnoble03** [comment]
@meetulr altho we should remove the commented out even if it's planned for the future. But, let's keep it as it is for now.

@palisadoes need one more approval, else good to go I think.

**Review by beingnoble03 — APPROVED** [review]

**Review by palisadoes — APPROVED** [review]

**Pull request merged**